pkg/reconciler/celltenant/interfaces.go

```diff
@@ -0,0 +1,227 @@
+/*
+Copyright 2021 Google LLC
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package celltenant
+
+import (
+	"fmt"
+
+	brokerv1beta1 "github.com/google/knative-gcp/pkg/apis/broker/v1beta1"
+	"github.com/google/knative-gcp/pkg/broker/config"
+	"github.com/google/knative-gcp/pkg/reconciler/broker/resources"
+	reconcilerutilspubsub "github.com/google/knative-gcp/pkg/reconciler/utils/pubsub"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	eventingduckv1beta1 "knative.dev/eventing/pkg/apis/duck/v1beta1"
+	"knative.dev/pkg/apis"
+)
+
+// DeletingTarget is the interface that the TargetReconciler takes for FinalizeKind().
+type DeletingTarget interface {
```

Comment:
Interface is normally a noun, maybe `TargetFinalizer` or `TargetCleaner`?

Reply:
How about `DeletableTarget` then? 

Actually do we need to introduce the interface just for the Finalizer at all? I am asking because:
1. Most fields are really for target not just for finalizer, eg `StatusUpdater` and `GetTopicID`. eg. SetStatusProjectID is likely not necessary due to StatusUpdater.
2. IIUC the future of Target is for Trigger and Subscription and they will both share the Finalizer, so it seems safe to me to use Target directly in the Finalizer too.

Reply:
Changed to `DeletableTarget`.

This PR doesn't need the interface, but it is useful once we add `Channel` support because the creating and deleting targets are distinct types, `SubscriberSpec` and `SubscriberStatus`, and have different fields. Looking at that area further, I probably could remove this interface if we want. The only meaningful difference between SubscriberSpec and SubscriberStatus for us is the inclusion of an optional DeliverySpec, which isn't used during deletion anyway.

Should I just remove it?

Reply:
Removed.